terraform/environments/data-platform/iam.tf

```
@@ -261,35 +461,28 @@ data "aws_iam_policy_document" "data_platform_product_bucket_policy_document" {
 
 # api gateway create data product metdata permissions
 data "aws_iam_policy_document" "iam_policy_document_for_create_metadata_lambda" {
+  source_policy_documents = [data.aws_iam_policy_document.log_to_bucket.json]
+
   statement {
-    sid     = "GetPutDataObject"
+    sid     = "GetPutMetadata"
     effect  = "Allow"
     actions = ["s3:GetObject", "s3:PutObject"]
     resources = [
-      "${module.s3-bucket.bucket.arn}/metadata/*",
-      "${module.s3-bucket.bucket.arn}/data_product_metadata_spec/*"
+      "${module.metadata_s3_bucket.bucket.arn}/metadata/*",
+      "${module.metadata_s3_bucket.bucket.arn}/data_product_metadata_spec/*"
     ]
   }
 
   statement {
-    sid    = "s3LogAccess"
-    effect = "Allow"
-    actions = [
-      "s3:GetObject",
-      "s3:PutObject",
-    ]
+    sid     = "ListBucket"
+    effect  = "Allow"
+    actions = ["s3:ListBucket"]
     resources = [
-      "${module.s3-bucket.bucket.arn}/logs/*"
+      module.metadata_s3_bucket.bucket.arn,
+      "${module.metadata_s3_bucket.bucket.arn}/*"
```

```suggestion
      module.metadata_s3_bucket.bucket.arn
```
This is a ListBucket operation, so only functions on the bucket object, not key paths